glossary: add shimpei, prairiedog and oci-add-hooks (#3202)
bottlerocket-os/bottlerocket: GLOSSARY.md

  It finds settings that need generation by way of metadata in the API, and calls helper programs specified by that metadata.
* [**thar-be-settings**](sources/api/thar-be-settings): A program that writes out system configuration files, replacing template variables with settings from the API.
* [**updog**](sources/updater/updog): An update client that interfaces with a specified TUF updates repository to upgrade or downgrade Bottlerocket hosts to different image versions.

> * [**prairiedog**](sources/api/prairiedog): A program provides support for various kernel boot related operations.

Reviewer: ```suggestion
* [**prairiedog**](sources/api/prairiedog): A program that handles various boot related operations.
```
